demo/b2share_demo/helpers.py

```diff
@@ -408,24 +451,62 @@ def _process_record(rec):
             'Text': 'Text',
             'Image': 'Image',
             'Other': 'Other',
+            'treebank':'Other',
+            'Time-Series':'Other'
         }
         result['resource_type'] = list(set([translate[r] for r in rec['resource_type']]))
-    #TODO fetch domain_metadata and translate to community specific metadata
-    #assume the current community specific block schema for metadata format
+    if 'domain_metadata' in rec.keys():
+        result.update(
+            _match_community_specific_metadata(rec, community)
+            )
     return result
 
+def _match_community_specific_metadata(rec, community):
+    cs_md_values_dict = rec['domain_metadata']
+    convert_to_array = [
+        'hasOntologyLanguage',
+        'usedOntologyEngineeringTool',
+    ]
+    for key in convert_to_array:
+        if key in cs_md_values_dict.keys():
+            cs_md_values_dict[key] = [
+                cs_md_values_dict[key]
+            ]
+    remove_integer_keys_with_empty_values = [
+        'planned_sampled_individuals',
+        'planned_total_individuals'
+    ]
+    for key in remove_integer_keys_with_empty_values:
+        if key in cs_md_values_dict.keys():
+            if cs_md_values_dict[key] == '':
+                 cs_md_values_dict.pop(key)
+            else:
+                cs_md_values_dict[key] = int(cs_md_values_dict[key])
+    result = {}
+    result['community_specific'] = {}
+    resolve_string = "$BLOCK_SCHEMA_ID[%s]" % community.name.lower()
+    block_schema_id = resolve_block_schema_id(resolve_string)
+    result['community_specific'][block_schema_id] = cs_md_values_dict
+    return result
 
 def _save_file(url, filename):
+    CHUNK_SIZE = 16 * 1024 * 1024 #download 16MB at a time
     current_app.logger.debug("Downloading %s" % url)
     gcontext = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
     finished = False
     f = open(filename, 'wb')
-    u = urlopen(url, context=gcontext)
-    counter = 0
+    try:
+        u = urlopen(url, context=gcontext)
+    except:
+        current_app.logger.error("TIMEOUT trying to download %s" % url)
+        pass
     while not finished:
         try:
-            f.write(u.read())
-            finished = True
+            chunk = u.read(CHUNK_SIZE)
+            if chunk:
+                f.write(chunk)
+            else:
+                finished = True
         except:
-            current_app.logger.debug("IncompleteRead? %d" % counter)
-            counter += counter
+            finished = True
```

Comment:
incorrect to assume that the file is correctly finished downloading, I'd simply quit with error message